### PR TITLE
Basic Gameplay

### PR DESCRIPTION
**Pull request by Nero-7**
Completed:
-Touch controls
-Interacting with hiding objects
-Can teleport between hiding objects
-Game loop completed (Timer, score, going from one scene to another...)

To-do:
-Fix animations when exiting hiding objects (can be fixed by remaking the animation, so that the slime doesn't need to move in the x-axis)
-Check the NPC sight, as when changing the game from x-y to x-z, some of the math needs some readjusting
-Check the guards, once the NPCs are fully operational, the guards shouldn't need to much tweaking
-Add sound effects
-Add visual effects
-Add leader boards
-Add functional options
-Integrate with Play Services

### CODE REVIEW

File: Slime Mall/Assets/Scripts/Managers/GameManager.cs [hunk 1]
```diff
@@ -16,10 +16,8 @@ public class GameManager : MonoBehaviour
     [Range(0, 360)][SerializeField][Tooltip("Length of initial game session")] float SessionLength = 120;
     [Tooltip("Score of current session")]  int Score = 0;
     [Tooltip("")] int HighScore = 0;
-    [Tooltip("How many points you gain per kill")] int KillScoreValue = 10;//SUGGESTION: Move it to the NPC's in case we have varying NPC's that offer different score
     [Tooltip("How much time you gain per kill")] int KillTimeValue = 1;//SUGGESTION: Move it to the NPC's in case we have varying NPC's that offer different time gain
```
Comment (Whizzo123) [inline Slime Mall/Assets/Scripts/Managers/GameManager.cs:19]: Think this is a good idea to future proof the code by having this inside the NPC class

Comment (Whizzo123) [inline Slime Mall/Assets/Scripts/Managers/GameManager.cs:19]: Why is the KillTimeValue in both GameManager and UI classes

File: Slime Mall/Assets/Scripts/Player/PlayerController.cs [hunk 2]
```diff
@@ -104,28 +94,21 @@ void Update()
...
-                    GameManager.Instance.UpdateScore();
-                    agent.enabled = true;
-                    targetNPC = null;
+                    KillNPC();
                 }
             }
 
-            else if(targetHS)
+            //Not hidden and has HS
+            else if((targetHS && !bIsHidden) || bIsTeleporting)
```
Comment (Whizzo123) [inline Slime Mall/Assets/Scripts/Player/PlayerController.cs:102]: What is the targetHS variabable?

File: Slime Mall/Assets/Scripts/Managers/GameManager.cs (hunk 1 above)
Comment (Nero-7) [reply Slime Mall/Assets/Scripts/Managers/GameManager.cs:19]: Left over code from the Jam

File: Slime Mall/Assets/Scripts/Player/PlayerController.cs (hunk 2 above)
Comment (Nero-7) [reply Slime Mall/Assets/Scripts/Player/PlayerController.cs:102]: HS means Hiding Spot, targetX refers to the current stored object of type X

**Review by Whizzo123 — APPROVED** [review]

**Pull request merged**